mountpoint-s3/src/data_cache/multilevel_cache.rs

```diff
@@ -0,0 +1,314 @@
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use futures::task::{Spawn, SpawnExt};
+use tracing::{trace, warn};
+
+use crate::object::ObjectId;
+
+use super::{BlockIndex, ChecksummedBytes, DataCache, DataCacheResult};
+
+pub struct MultilevelDataCache<DiskCache, ExpressCache, Runtime> {
+    disk_cache: Arc<DiskCache>,
+    express_cache: ExpressCache,
+    runtime: Runtime,
+}
+
+impl<DiskCache: DataCache, ExpressCache: DataCache, Runtime: Spawn>
+    MultilevelDataCache<DiskCache, ExpressCache, Runtime>
+{
+    pub fn new(disk_cache: Arc<DiskCache>, express_cache: ExpressCache, runtime: Runtime) -> Self {
+        assert_eq!(disk_cache.block_size(), express_cache.block_size());
```

Comment:
Why does this matter?

Reply:
There is a method [block_size](https://github.com/awslabs/mountpoint-s3/pull/1064/files/6a8e202299cab80ee35bb6dac238dc1d2d8f6cba#diff-9106dd30662c843c8114dbe0453395ba43b722d8777fda494f020e820e8b1644R94) below, which's implemented based on this assumption. Later this method is used in CachingPartStream on [reads](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/prefetch/caching_stream.rs#L135) from cache and on [writes](https://github.com/awslabs/mountpoint-s3/blob/main/mountpoint-s3/src/prefetch/caching_stream.rs#L196) to it.